app/(auth)/signup/page.tsx

```diff
@@ -0,0 +1,131 @@
+'use client';
+
+import { useState } from 'react';
+import PasswordComplexity from '@/app/utils/PasswordComplexity';
+import PasswordInput from '@/app/utils/PasswordInput';
+import { AuthProvider, useAuth } from '../../utils/AuthProvider';
+
+export default function SignUpLayout() {
+  return (
+    <AuthProvider>
+      <SignUp />
+    </AuthProvider>
+  );
+}
+
+function SignUp() {
+  const { signUp } = useAuth();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState<string | null>(null);
+  const [confirmPassword, setConfirmPassword] = useState<string | null>(null);
+  const [passwordError, setPasswordError] = useState('');
+  const [passwordComplexityError, setPasswordComplexityError] = useState<
+    string | null
+  >(null);
+  const [passwordComplexity, setPasswordComplexity] = useState<boolean>(false);
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+
+  // Handles input to password
+  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newPassword = e.target.value;
+    setPassword(newPassword);
+    validatePasswords(newPassword, confirmPassword);
+    validatePasswordComplexity(newPassword);
+  };
+
+  // Handles input to confirm password
+  const handleConfirmPasswordChange = (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const newConfirmPassword = e.target.value;
+    setConfirmPassword(newConfirmPassword);
+    validatePasswords(password, newConfirmPassword);
+  };
+
+  // Checks if passwords match and sets error
+  const validatePasswords = (
+    password: string | null,
+    confirmPassword: string | null,
+  ) => {
+    if (password !== confirmPassword) {
+      setPasswordError('Passwords do not match.');
+    } else {
+      setPasswordError(''); // Clear error when passwords match
+    }
+  };
+
+  // Set password complexity error if requirements are not met
+  const validatePasswordComplexity = (password: string | null) => {
+    const hasLowerCase = /[a-z]/.test(password || '');
+    const hasNumber = /\d/.test(password || '');
+    const longEnough = (password || '').length >= 8;
+
+    if (password && hasLowerCase && hasNumber && longEnough) {
+      setPasswordComplexity(true);
+      setPasswordComplexityError(null); // Clear error if all conditions are met
+    } else if (password) {
+      setPasswordComplexity(false);
+      setPasswordComplexityError('Password must meet complexity requirements');
```

Comment:
Just flagging that it makes more sense for this error to only show up if user tries to submit without meeting requirements. it's fine for now but we may want to adjust the logic in the future